src/snowflake/connector/crl.py

```
@@ -0,0 +1,576 @@
+            # Check if CRL is expired
+            try:
+                next_update = crl.next_update_utc
+            except AttributeError:
+                next_update = crl.next_update
+
+            if next_update and now > next_update:
+                logger.warning(
+                    "The CRL from %s was expired on %s", crl_url, next_update
+                )
+                return None, None
+
+            return crl, now
+        except Exception:
+            logger.exception("Failed to parse CRL from %s", crl_url)
+            return None, None
+
+    def _check_certificate_against_crl_url(
+        self, cert: x509.Certificate, ca_cert: x509.Certificate, crl_url: str
+    ) -> CRLValidationResult:
+        """Check if certificate is revoked according to CRL by the provided URL"""
+        now = datetime.now(timezone.utc)
+        logger.debug("Trying to get cached CRL for %s", crl_url)
+        cached_crl = self._get_crl_from_cache(crl_url)
+        if (
+            cached_crl is None
+            or cached_crl.is_crl_expired_by(now)
+            or cached_crl.is_evicted_by(now, self._cache_validity_time)
+        ):
+            crl, ts = self._download_crl(crl_url)
+            if crl and ts:
+                self._put_crl_to_cache(crl_url, crl, ts)
+        else:
+            crl = cached_crl.crl
+
+        # If by some reason we didn't get a valid CRL we consider it a check error
+        if crl is None:
+            return CRLValidationResult.ERROR
+
+        # Verify CRL signature with CA public key
+        # Check if the CA certificate is the expected CRL issuer
+        if crl.issuer != ca_cert.subject:
+            logger.warning(
+                "CRL issuer (%s) does not match CA certificate subject (%s) for URL: %s",
+                crl.issuer,
+                ca_cert.subject,
+                crl_url,
+            )
+            # In most cases this indicates a configuration issue, but we'll still try verification
```

We should return an error here. This is specifically part of the RFC 5280 Section 6.3 algorithm.
It is extremely unlikely that a CA use the same key with different "issuers", but it is possible. 

(Certainly 12 years ago there were several examples of key reuse across different issuers. It would be much more difficult to do today given the recent "key freshness" requirements for new CAs and there might even be an explicit restriction that I just glossed over as the requirement never pertained to me, but we should error out regardless.)